pkg/datasources/external_functions.go

```diff
@@ -56,42 +58,59 @@ var externalFunctionsSchema = map[string]*schema.Schema{
 
 func ExternalFunctions() *schema.Resource {
 	return &schema.Resource{
-		Read:   ReadExternalFunctions,
-		Schema: externalFunctionsSchema,
+		ReadContext: ReadContextExternalFunctions,
+		Schema:      externalFunctionsSchema,
 	}
 }
 
-func ReadExternalFunctions(d *schema.ResourceData, meta interface{}) error {
+func ReadContextExternalFunctions(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	db := meta.(*sql.DB)
+	client := sdk.NewClientFromDB(db)
 	databaseName := d.Get("database").(string)
 	schemaName := d.Get("schema").(string)
 
-	currentExternalFunctions, err := snowflake.ListExternalFunctions(databaseName, schemaName, db)
-	if errors.Is(err, sql.ErrNoRows) {
-		// If not found, mark resource to be removed from state file during apply or refresh
-		log.Printf("[DEBUG] external functions in schema (%s) not found", d.Id())
-		d.SetId("")
-		return nil
-	} else if err != nil {
-		log.Printf("[DEBUG] unable to parse external functions in schema (%s)", d.Id())
+	req := sdk.NewShowExternalFunctionRequest()
+	externalFunctions, err := client.ExternalFunctions.Show(ctx, req)
+	if err != nil {
 		d.SetId("")
 		return nil
 	}
 
-	externalFunctions := []map[string]interface{}{}
-
-	for _, externalFunction := range currentExternalFunctions {
+	externalFunctionsList := []map[string]interface{}{}
+	for _, externalFunction := range externalFunctions {
 		externalFunctionMap := map[string]interface{}{}
+		externalFunctionMap["name"] = externalFunction.Name
+
+		// do we filter by database?
+		currentDatabase := strings.Trim(externalFunction.CatalogName, `"`)
+		if databaseName != "" {
+			if currentDatabase != databaseName {
+				continue
+			}
+			externalFunctionMap["database"] = currentDatabase
+		} else {
+			externalFunctionMap["database"] = currentDatabase
+		}
 
-		externalFunctionMap["name"] = externalFunction.ExternalFunctionName.String
-		externalFunctionMap["database"] = externalFunction.DatabaseName.String
-		externalFunctionMap["schema"] = externalFunction.SchemaName.String
-		externalFunctionMap["comment"] = externalFunction.Comment.String
-		externalFunctionMap["language"] = externalFunction.Language.String
+		// do we filter by schema?
+		currentSchema := strings.Trim(externalFunction.SchemaName, `"`)
+		if schemaName != "" {
+			if currentSchema != schemaName {
+				continue
+			}
+			externalFunctionMap["schema"] = currentSchema
+		} else {
+			externalFunctionMap["schema"] = currentSchema
+		}
 
-		externalFunctions = append(externalFunctions, externalFunctionMap)
+		externalFunctionMap["comment"] = externalFunction.Description
+		externalFunctionMap["language"] = externalFunction.Language
+		externalFunctionsList = append(externalFunctionsList, externalFunctionMap)
 	}
 
-	d.SetId(fmt.Sprintf(`%v|%v`, databaseName, schemaName))
-	return d.Set("external_functions", externalFunctions)
+	d.SetId(fmt.Sprintf(`external_functions|%v|%v`, databaseName, schemaName))
```

Comment:
why the identifier is different, will it cause issues for users who already use it? Because I don't think we need this prefix

Reply:
anyways i changed it back to what it was